extensions/git/src/git.ts

```diff
@@ -1536,6 +1536,25 @@ export class Repository {
 		await this.exec(args);
 	}
 
+	async fastForwardBranch(options: { remote: string; ref: string; readonly cancellationToken?: CancellationToken }): Promise<void> {
+		const args = ['fetch', options.remote, `${options.ref}:${options.ref}`];
+		const spawnOptions: SpawnOptions = {
+			cancellationToken: options.cancellationToken,
+			env: { 'GIT_HTTP_USER_AGENT': this.git.userAgent }
```

Comment:
I wonder why we don't set this in the `exec` util for all commands... 🤔 Might be a future cleanup task.